astropy/nddata/covariance.py

```diff
@@ -0,0 +1,1034 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+Defines a class used to store and interface with covariance matrices.
+"""
+
+import warnings
+from pathlib import Path
+
+import numpy as np
+from matplotlib import pyplot as plt
+from scipy import sparse
+
+from astropy import log
+from astropy.io import fits
```

Comment:
Can you say more about this, or point me to an example file?  I tried just adopting this edit and the code complains about `sparse` not being available.  I expect I'm missing something obvious.

Reply:
The not quite so obvious complement is to add the conditional import just around the parts of the code using `scipy.sparse`, as in
https://github.com/astropy/astropy/blob/2c17013da0e5a03f89f8bf51aa06e7d8445850d9/astropy/convolution/convolve.py#L58-L60
but since in the current implementation I guess this would be the entire `Covariance` class, need to think of a graceful way to just define that as a `NotImplemented` or similar otherwise.

Reply:
Got it.  I've implemented something similar to what's in `astropy/cosmology/flrw/lambdacdm.py`.  Let me know what you think.